Tests/Algorithm/Framework/Portfolio/RiskParityPortfolioConstructionModelTests.cs

```diff
@@ -0,0 +1,227 @@
+            }
+        }
+
+        [Test]
+        public void NewtonMethodOptimizationTest()
+        {
+            var testOptimizer = new TestRiskParityPortfolioOptimizer();
+
+            Func<double[], double> objective = (x) => 1/2 * Matrix.Dot(Matrix.Dot(x, _covariance1), x) - Matrix.Dot(_expectedReturn, x);
+            Func<double[], double[]> jacobian = (x) => Elementwise.Subtract(Matrix.Dot(_covariance1, x), _expectedReturn);
+            Func<double[], double[,]> hessian = (x) => _covariance1;
+
+            var result = testOptimizer.TestNewtonMethodOptimization(1, objective, jacobian, hessian);
+            Assert.AreEqual(new double[]{1d}, result);
+
+            var exception = Assert.Throws<ArgumentException>(() => testOptimizer.TestNewtonMethodOptimization(0, objective, jacobian, hessian));
+            Assert.That(exception.Message, Is.EqualTo("Argument \"numOfVar\" must be a positive integer between 1 and 1000"));
+
+            exception = Assert.Throws<ArgumentException>(() => testOptimizer.TestNewtonMethodOptimization(2000, objective, jacobian, hessian));
+            Assert.That(exception.Message, Is.EqualTo("Argument \"numOfVar\" must be a positive integer between 1 and 1000"));
+
+            result = testOptimizer.TestNewtonMethodOptimization(2, objective, jacobian, hessian);
+            result = result.Select(x => Math.Round(x, 2)).ToArray();
+            Assert.AreEqual(_expectedResult1, result);
+            
+            objective = (x) => 1/2 * Matrix.Dot(Matrix.Dot(x, _covariance2), x) - Matrix.Dot(_expectedReturn, x);
+            jacobian = (x) => Elementwise.Subtract(Matrix.Dot(_covariance2, x), _expectedReturn);
+            hessian = (x) => _covariance2;
+
+            result = testOptimizer.TestNewtonMethodOptimization(2, objective, jacobian, hessian);
+            result = result.Select(x => Math.Round(x, 2)).ToArray();
+            Assert.AreEqual(_expectedResult2, result);
+            
+            objective = (x) => 1/2 * Matrix.Dot(Matrix.Dot(x, _covariance3), x) - Matrix.Dot(_expectedReturn, x);
+            jacobian = (x) => Elementwise.Subtract(Matrix.Dot(_covariance3, x), _expectedReturn);
+            hessian = (x) => _covariance3;
+
+            result = testOptimizer.TestNewtonMethodOptimization(2, objective, jacobian, hessian);
+            result = result.Select(x => Math.Round(x, 2)).ToArray();
+            Assert.AreEqual(_expectedResult3, result);
+            
+            objective = (x) => 1/2 * Matrix.Dot(Matrix.Dot(x, _covariance4), x) - Matrix.Dot(_expectedReturn, x);
+            jacobian = (x) => Elementwise.Subtract(Matrix.Dot(_covariance4, x), _expectedReturn);
+            hessian = (x) => _covariance4;
+
+            result = testOptimizer.TestNewtonMethodOptimization(2, objective, jacobian, hessian);
+            result = result.Select(x => Math.Round(x, 2)).ToArray();
+            Assert.AreEqual(_expectedResult4, result);
```

Comment:
These different cases look like the same test with different arguments maybe?

I think should add at least a couple unit tests for the `RiskParityPortfolioOptimizer` & `RiskParityPortfolioOptimizer` themselves, in their own test suite?